middle_end/base_types/closure_origin.ml

```diff
@@ -0,0 +1,22 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
```

Comment:
Who is the author of this patch? Did you copy this file from something written by Pierre and Mark?

Reply:
This should have "Fu Yong Quah, Jane Street Europe" in the header as well.  I wrote some of this originally.  We should maybe have a more generic header on the Flambda files which points at an "authors" file.